miasm2/expression/expression.py

```diff
@@ -850,23 +896,30 @@ def __init__(self, args):
         for e, a, b in args:
             assert(a >= 0 and b >= 0)
             o.append(tuple([e, a, b]))
-        self.args = tuple(o)
+        self.__args = tuple(o)
+
+        self.__size = max([x[2]
```

Comment:
Thanks to your patches, `ExprCompose` are now immutable. Therefore, the `args` attribute is immutable too, and set only during the `__init__`.

That way, `args` attribute could be ordered (non-overlapping intervals). And then, this code snippet could be re-write:
```Python
self.__size = self.args[-1][-1] - self.args[0][0]
```

In addition, we can now improve the comparison between `ExprCompose`, because we currently have:
```Python
from miasm2.expression.expression import ExprCompose, ExprInt32
e1 = ExprCompose([(ExprInt32(1), 0, 32), (ExprInt32(2), 32, 64)])
e2 = ExprCompose([(ExprInt32(1), 0, 32), (ExprInt32(2), 32, 64)][::-1])

e1 == e2
# False, but should be True
```

It is a direct consequence of your patch, so in my opinion, you can add it in this PR.